src/yang_xpath.erl

```diff
@@ -77,7 +77,11 @@ compile(Str, Pos, M, Strict, Ctx0) ->
             Prefixes = xpath_rewrite:fold_expr(fun prefix/2, [], Q),
             Ctx2 = lists:foldl(
                      fun(P, Ctx1) ->
-                             yang:mark_import_as_used(Name, P, Ctx1)
+                             %% The prefix must be marked as used in
+                             %% the target module also.
```

Comment:
Why is this additional code added?   The test case you have added isn't affected by this code.

Reply:
Ah, this is because a test case that checks dependencies (not touched by the commit) fails unless this is fixed.
I'll include that test case too in the commit.

Also, a lot of the testing was done in eunit, does eunit cases run in this repo?

Reply:
If the failing dependency test is unrelated to this PR, please add that test and that fix in a separate PR.

As for eunit, no, there are no eunit tests yet.  But if you have such tests, please add them!

Nowadays I'm using erlang.mk for all my projects, perhaps I should change yanger as well.  Then it would be easier to run eunit etc.  Would a switch to erlang.mk be a problem for you?

Reply:
Well, it's not exactly unrelated to this PR, because this PR changes slightly how identifiers are resolved during deviation. This will then incur that plugins, in our case the fxs plugin, will do a slightly different output unless care is taken (as is done in yanger_fxs).
As it turns out, our failing test is not even in yanger so appears to be out of scope for this PR.

I'll add the eunit tests, the xpath case is covered there.

Should I remove the yanger_xpath from the commit or keep it in?
I would prefer to keep it in.

Reply:
Are you saying that w/o this PR, the xpath-related test works fine w/o the yanger_xpath fix?  If this is the case I need to take a closer look.  My gut feeling is that the fix in yanger_xpath is not correct.

Reply:
Yes, the test case passes if PR is not present.

Let me include the yang code of the test case:
module test-remote-deviation-import-data {
  namespace "http://test-data";
  prefix td;
  list item1 {
    key name;
    leaf name {
      type string;
    }
    leaf value {
      type uint8;
    }
  }
}

module test-remote-deviation-import.yang.deviation {
  namespace "http://test-devn";
  prefix test-devn;
  import test-remote-deviation-import {prefix t;}
  import test-remote-deviation-import-data {prefix td;}
  import d {prefix d;}
  
  deviation /t:item/t:name {
    deviate replace {
      type leafref {
        path "/td:item1/td:name";
      }
    }
  }
}

So in the deviating module (test-remote-deviation-import.yang.deviation) there's a leafref.
In legacy, this leafref is resolved/validated in target module scope (test-remote-deviation-import-data) and therefore the mark_import_as_used works fine (for the test case, the prefix is not marked for use in deviating module which it perhaps should)

In the PR, the leafref is resolved/validated in deviating module scope, which I think is correct, but that also means it will only mark the prefix in deviating module, missing the target module.

Reply:
Can you update the PR with the test?

I used your modules above, and created a module `test-remote-deviation-import`, and removed your fix to `yang_xpath.erl`, and it seems to work:
```
$ anger test-remote-deviation-import.yang  --deviation-module test-remote-deviation-import-dev.yang  -f tree
module: test-remote-deviation-import
  +--rw item* [name]
     +--rw name     -> /td:item1/name
     +--rw value?   uint8
```

Reply:
Ah, I can see that the test code actually is degenerated, my bad.
If we imagine that the prefix td points to third module, then it would not be marked as used in test-remote-deviation-import-data. Hmm, if we imagine that test-remote-deviation-import-data imported that *then* it would not be noted as used. Lots of imagination here, let me come up with better sample code.